### PR TITLE
Fix for vsix package not finding some dependencies on activation

### PR DESCRIPTION
**Pull request by ngondalia**
- Used webpack to package the extension
- Added additional options to TS compiler config

### CODE REVIEW

**Comment by ngondalia** [comment]
Questions for the devs.. Is there a licensing file i should put anywhere that includes the copyright for blackberry?

File: common-tsconfig.json [hunk 1]
```diff
@@ -17,6 +17,9 @@
         "strict": true,
         "sourceMap": true,
         "strictPropertyInitialization": false,
-        "target": "ES2017"
+        "target": "ES2017",
+        "esModuleInterop": true,
+        "baseUrl": ".",
+        "composite": true
     }
 }
```
Comment (bhufmann) [inline common-tsconfig.json:25]: I get a marker that a newline  is missing at the end of file. Could please add it?

File: vscode-trace-extension/.vscodeignore [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.vscode
+node_modules
+tsconfig.json
+webpack.config.js
+*.tsbuildinfo
+.eslintrc.js
+src
```
Comment (bhufmann) [inline vscode-trace-extension/.vscodeignore:7]: add newline at the end of file

File: vscode-trace-extension/tsconfig.json [hunk 3]
```diff
@@ -1,10 +1,19 @@
...
+    "exclude": [
+        "node_modules",
+        "lib"
+    ],
+    "references": [
+        {
+            "path": "../vscode-trace-webviews"
+        }
     ]
 }
```
Comment (bhufmann) [inline vscode-trace-extension/tsconfig.json:19]: newline at the end of file

File: vscode-trace-extension/webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+                            limit: 10000,
+                        },
+                    },
+                ],
+            }
+        ]
+    },
+}
+
+module.exports = config;
```
Comment (bhufmann) [inline vscode-trace-extension/webpack.config.js:70]: newline at the end of file

**Review by bhufmann — CHANGES_REQUESTED** [review]
Thanks for the contribution. It works and I'm able to install the vsix extension to VsCode without an error.

I have some minor comments. Please also update your git commit message:

- Have a title for the commit
- Describe the change (you can use your bullet points)
- What it actually fixes
- Add a signed-off-by: Signed-off-by: FirstName LastName <email address>

**Comment by bhufmann** [comment]
I noticed a problem when running it from development environment. Using the launch configuration `Launch Extension` (press `F5`) to start vscode with the trace viewer extension in debug mode, I get the error message below when switching to the `trace viewer` view. Could you please fix it with your changes? Thanks!

image

**Comment by ngondalia** [comment]
> 

The error you are seeing is likely due to webpack script was not ran. I will add it under prepare script so it gets ran by default when you run yarn.

**Comment by ngondalia** [comment]
> Questions for the devs.. Is there a licensing file i should put anywhere that includes the copyright for blackberry?

Any input on this?

**Comment by bhufmann** [comment]
> > Questions for the devs.. Is there a licensing file i should put anywhere that includes the copyright for blackberry?
> 
> Any input on this?

I would suggest the LICENSE file in the root directory of the repository. WDYT?

**Review by bhufmann — APPROVED** [review]
It looks good to me. Thanks a lot for this fix. Do you want me to merge it or do you want to update the License file in this PR first?

**Comment by ngondalia** [comment]
> It looks good to me. Thanks a lot for this fix. Do you want me to merge it or do you want to update the License file in this PR first?

Let me update the license file first, and then you can merge it.

**Comment by ngondalia** [comment]
> > > Questions for the devs.. Is there a licensing file i should put anywhere that includes the copyright for blackberry?
> > 
> > 
> > Any input on this?
> 
> I would suggest the LICENSE file in the root directory of the repository. WDYT?

I have update the license file. Let me know if that works for you. Thanks!

**Review by MatthewKhouzam — APPROVED** [review]
Looks great to me! Thanks!

**Pull request merged**